### PR TITLE
feat: added product listing page

### PR DESCRIPTION
**Pull request by jDineshKarthic**
This branch contains the code for product page.

### CODE REVIEW

**Review by aliamaanansari — APPROVED** [review]
gone through every line i cant find any thing odd as per my knowledge , UI looks good code's clean and usable! great workkkk !

File: components/product-listing/product.css [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    padding-right: 4rem;
+    border-radius: 1rem;
+    border: none;
+    /* color: var(--dk-grey); */
+    box-shadow: 0 1px 8px #00000040;
+    width: 30%;
+    font-weight: 500;
+    padding-left: 1.5rem;
+    border-style: none;
+    outline: none;
```
Comment (ttahm3d) [inline components/product-listing/product.css:60]: Outline none creates an accessibility issue. 

Instead do

```css
outline: 2px solid transparent;
outline-offset: 2px;
```

File: components/product-listing/product.css [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+}
+
+.input-items {
+    padding: 0.6rem;
+    padding-right: 4rem;
+    border-radius: 1rem;
+    border: none;
+    /* color: var(--dk-grey); */
+    box-shadow: 0 1px 8px #00000040;
+    width: 30%;
```
Comment (ttahm3d) [inline components/product-listing/product.css:56]: width:30% will look too small on mobile screens

**Review by neetukumari4858 — APPROVED** [review]
overall looks good just one suggestion is that add link for visualising.

**Review by ttahm3d — COMMENTED** [review]
Looks good. Have added a comment please take a look

File: components/product-listing/product.css (hunk 1 above)
Comment (jDineshKarthic) [reply components/product-listing/product.css:60]: ok thahir i'll change it

**Review by JantuDeb — APPROVED** [review]
CSS and HTML looking good. Work on the functionality use `Link` from react-router instead of `<a>`,  put all the categories in an `array` then use `map()` to get all the `checkbox`.

**Pull request merged**